subsystem/ipv6_test.go

```diff
@@ -0,0 +1,262 @@
+package subsystem
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/units"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/models"
+)
+
+var (
+	validHwInfoV6 = &models.Inventory{
+		CPU:    &models.CPU{Count: 16},
+		Memory: &models.Memory{PhysicalBytes: int64(32 * units.GiB)},
+		Disks: []*models.Disk{
+			{DriveType: "SSD", Name: "loop0", SizeBytes: validDiskSize},
+			{DriveType: "HDD", Name: "sdb", SizeBytes: validDiskSize}},
+		Interfaces: []*models.Interface{
+			{
+				IPV6Addresses: []string{
+					"1001:db8::10/120",
+				},
+			},
+		},
+		SystemVendor: &models.SystemVendor{Manufacturer: "manu", ProductName: "prod", SerialNumber: "3534"},
+		Timestamp:    1601853088,
+	}
+)
+
+var _ = Describe("IPv6 installation", func() {
+	var (
+		ctx         = context.Background()
+		cluster     *models.Cluster
+		clusterCIDR = "2002:db8::/53"
+		serviceCIDR = "2003:db8::/112"
+		clusterID   strfmt.UUID
+	)
+
+	AfterEach(func() {
+		clearDB()
+	})
+	BeforeEach(func() {
+		registerClusterReply, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
+			NewClusterParams: &models.ClusterCreateParams{
+				BaseDNSDomain:            "example.com",
+				ClusterNetworkCidr:       &clusterCIDR,
+				ClusterNetworkHostPrefix: 64,
+				Name:                     swag.String("test-cluster"),
+				OpenshiftVersion:         swag.String("4.6"),
+				PullSecret:               swag.String(pullSecret),
+				ServiceNetworkCidr:       &serviceCIDR,
+				SSHPublicKey:             "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC50TuHS7aYci+U+5PLe/aW/I6maBi9PBDucLje6C6gtArfjy7udWA1DCSIQd+DkHhi57/s+PmvEjzfAfzqo+L+/8/O2l2seR1pPhHDxMR/rSyo/6rZP6KIL8HwFqXHHpDUM4tLXdgwKAe1LxBevLt/yNl8kOiHJESUSl+2QSf8z4SIbo/frDD8OwOvtfKBEG4WCb8zEsEuIPNF/Vo/UxPtS9pPTecEsWKDHR67yFjjamoyLvAzMAJotYgyMoxm8PTyCgEzHk3s3S4iO956d6KVOEJVXnTVhAxrtLuubjskd7N4hVN7h2s4Z584wYLKYhrIBL0EViihOMzY4mH3YE4KZusfIx6oMcggKX9b3NHm0la7cj2zg0r6zjUn6ZCP4gXM99e5q4auc0OEfoSfQwofGi3WmxkG3tEozCB8Zz0wGbi2CzR8zlcF+BNV5I2LESlLzjPY5B4dvv5zjxsYoz94p3rUhKnnPM2zTx1kkilDK5C5fC1k9l/I/r5Qk4ebLQU= oscohen@localhost.localdomain",
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		cluster = registerClusterReply.GetPayload()
+		clusterID = *cluster.ID
+		log.Infof("Register cluster %s", cluster.ID.String())
+	})
+	It("[only_k8s]report_host_progress", func() {
```

Comment:
This test logic doesn't need to be verified twice (it's been copied from `cluster_test`). You should minimize it to only IPv6 related stuff.

Reply:
What do you mean by "only IPv6 related stuff"?

Reply:
This test for example checks various "report" scenarios. It isn't related to IPv4 nor IPv6.
You should just verify an installation or specific edge-cases where you think it might be relevant.

Reply:
Actually,  there are no known edge cases specific to IPv6 that should be tested here.  Just happy flow.

Reply:
Left just happy flow